src/overlay/SurveyMessageLimiter.cpp

```diff
@@ -139,6 +140,56 @@ SurveyMessageLimiter::recordAndValidateResponse(
     return true;
 }
 
+bool
+SurveyMessageLimiter::validateStartSurveyRecording(
+    TimeSlicedSurveyStartRecordingMessage const& startSurvey,
+    std::function<bool()> onSuccessValidation)
+{
+    if (!surveyLedgerNumValid(startSurvey.ledgerNum))
+    {
+        // Request too old (or otherwise invalid)
+        return false;
+    }
+
+    if (mApp.getSurveyDataManager().surveyIsActive())
+    {
+        // Survey already active, toss
+        return false;
+    }
+
+    if (!onSuccessValidation())
+    {
+        return false;
+    }
+
+    // Unlike survey versions 0 and 1, we do not need to check for too many
```

Comment:
Keeping track of number of surveys in flight seems like something that should be managed by the limiter class, rather than the data manager. Perhaps  SurveyManager can pass whatever information is needed (queried from SurveyDataManager) for the limiter to make the call on validity.

Reply:
In this case I just wrote a bad comment. The `surveyDataManager.surveyIsActive()` check is the `SurveyMessageLimiter` checking whether a survey is already running. I updated the comment in 8c26afb71972a31b64013cdf39dde53ca84d2327 to make this clearer.